solr/core/src/test/org/apache/solr/core/CoreSorterTest.java

```diff
@@ -187,7 +188,7 @@ private CoreDescriptor newCoreDescriptor(Replica r) {
     Map<String,String> props = map(
         CoreDescriptor.CORE_SHARD, r.getShard(),
         CoreDescriptor.CORE_COLLECTION, r.getCollection(),
-        CoreDescriptor.CORE_NODE_NAME, r.getNodeName()
+        CommonParams.REPLICA_NAME, r.getNodeName()
```

Comment:
r.getNodeName() should return the name of the node that the replica resides on. Use r.getName()